pkg/tsdb/cloudmonitoring/cloudmonitoring.go

```diff
@@ -221,16 +232,48 @@ func (e *CloudMonitoringExecutor) executeTimeSeriesQuery(ctx context.Context, ts
 			break
 		}
 		query.Params.Set("resourceType", resourceType)
-		dl := ""
-		if len(resp.TimeSeries) > 0 {
-			dl = query.buildDeepLink()
+
+		err = e.parseResponse(queryRes, resp, query)
+		if err != nil {
+			queryRes.Error = err
 		}
-		queryRes.Meta.Set("deepLink", dl)
+
+		if len(unit) > 0 {
+			frames, _ := queryRes.Dataframes.Decoded()
```

Comment:
the only problem of the parseresponse, it is that it doesn't have access to all request. If we check the logics on frontend, we see whether all requests have the same unit, if not, we don't set the unit into dataframe (timeseries), only if every unit is the same, we set it in the response. that is the reason why the unit is set here, but not the deeplink

Reply:
Good points! Decoding can be expensive and it's better to be avoided but checking closer I think it's fine as is: `parseResponse()` calls  `NewDecodedDataFrames()` for creating the Dataframes which actually stores the data decoded:
https://github.com/grafana/grafana/blob/dbfee130b1e1c0e84cac20020eff086ee488ca17/pkg/tsdb/models.go#L218-L222

Then, `Dataframes.Decoded()` since data are decoded just returns them without preforming anything expensive:
https://github.com/grafana/grafana/blob/dbfee130b1e1c0e84cac20020eff086ee488ca17/pkg/tsdb/models.go#L248-L258

WDYT?